apps/api/src/app/events/usecases/message-matcher/message-matcher.usecase.ts

```diff
@@ -225,6 +238,115 @@ export class MessageMatcher {
     ));
   }
 
+  private async processHaveSeen(
+    filter: IPreviousStepFilterPart,
+    command: SendMessageCommand,
+    filterProcessingDetails: FilterProcessingDetails
+  ) {
+    const previousJobId = command.job._parentId;
+
+    if (!previousJobId) {
+      return true;
+    }
+    const message = await this.messageRepository.findOne({
+      _jobId: previousJobId,
+      _organizationId: command.organizationId,
+      _environmentId: command.environmentId,
+    });
+
+    if (!message) {
+      return true;
+    }
+    if (message?.channel === ChannelTypeEnum.IN_APP) {
+      filterProcessingDetails.addCondition({
+        filter: FILTER_TO_LABEL[filter.on],
+        field: 'read',
```

Comment:
```suggestion
        field: 'seen',
```
Not sure about this, though.